regolith/helpers/a_proposalhelper.py

```diff
@@ -0,0 +1,190 @@
+
+       A proposal is a dictionary object describing a research or
+        project proposal submitted by the group.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = "a_proposal"
+    needed_dbs = [f'{TARGET_COLL}', 'people', 'groups']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        #rc.coll = f"{rc.target_collection}"
+        if not rc.database:
+            rc.database = rc.databases[0]["name"]
+        gtx[rc.coll] = sorted(
+            all_docs_from_collection(rc.client, rc.coll), key=_id_key
+        )
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def db_updater(self):
+        gtx = self.gtx
+        rc = self.rc
+        now = dt.date.today()
+        key = f"{str(now.year)[2:]}_{''.join(rc.name.casefold().split()).strip()}"
+
+        coll = self.gtx[rc.coll]
+        pdocl = list(filter(lambda doc: doc["_id"] == key, coll))
+        if len(pdocl) > 0:
+            raise RuntimeError(
+                "This entry appears to already exist in the collection")
+        else:
+            pdoc = {}
+        pdoc.update({'_id': key,
+                     'amount': float(rc.amount)
+                     })
+        if isinstance(rc.authors, str):
+            pdoc.update({'authors': [rc.authors]})
+        else:
+            pdoc.update({'authors': rc.authors})
+        if rc.begin_date != 'tbd':
+            pdoc.update({'begin_date': date_parser.parse(rc.begin_date).date()})
+        else:
+            pdoc.update({'begin_date': rc.begin_date})
+        cpp_info = {}
+        if rc.cppflag:
+            cpp_info['cppflag'] = True
+        else:
+            cpp_info['cppflag'] = False
+        cpp_info['other_agencies_submitted'] = rc.other_agencies
+        cpp_info['institution'] = rc.institution
+        cpp_info['person_months_academic'] = int(rc.months_academic)
+        cpp_info['person_months_summer'] =  int(rc.months_summer)
+        cpp_info['project_scope'] = rc.scope
+        pdoc.update({'cpp_info':cpp_info})
+        pdoc.update({'currency': rc.currency})
+        if rc.due_date != 'tbd':
+            pdoc.update({'due_date': date_parser.parse(rc.due_date).date()})
+        else:
+            pdoc.update({'due_date': rc.due_date})
+        if rc.duration != 'tbd':
+            pdoc.update({'duration': int(rc.duration)})
+            if rc.begin_date != 'tbd':
+                begin_date = date_parser.parse(rc.begin_date).date()
+                pdoc.update({'end_date': begin_date + relativedelta(months = int(rc.duration))})
+            else:
+                pdoc.update({'end_date': 'tbd'})
+        else:
+            pdoc.update({'duration': rc.duration,
```

Comment:
is duration in the schema or just end_date?  It seems (as we have discussed before) that there is redundancy here.  Maybe we should think about this.

Reply:
`duration` is in the schema explicitly – but I was wondering this too. I think it might be helpful to maintain it for list helpers (for e.g. all proposals of `duration` `x`) to avoid the extra computation, or in a situation where we know the `duration` but not the `begin_date`

Reply:
not sure how likely it is that such a situation would arise though

Reply:
fixed!